app/models/event.rb

```diff
@@ -22,4 +22,10 @@ class Event < ApplicationRecord
   def service_description
     service_category.service_category_name
   end
+
+  def exception_notes(zip_code)
```

Comment:
Is this just a little bit of logic that only puts the exception not in if the zip code it matches the searched for zip code?

Reply:
TL;DR 

Yes
++++++

exception_notes(zip_code)  is the Event instance method.  It is called by the event_serializer which gets the submitted zip_code from the agency_controller through an instance_option.  The serializer calls this method on the Event object (instance) it is decorating:

exception_note = object.exception_notes(@instance_options[:zip_code])

Then the event instance method makes this call

event_zip = event_zip_codes.find_by(zip_code: zip_code)

Because the event model has the association "has_many event_zip_codes" the lookup is limited to the event_zip_codes for the event object.   So now we have the event_service_geography record whose geo_value, alias zip_code, matches the submitted zip code.  

The cool thing about Nikhil's first version was that the serializer was not involved.  I think during the query maybe as each event instance was added to the active record results the instance method was getting invoked or something like that.

The final step is event_zip.exception_note which is an instance method in the event_zip_code model and it returns the exception note up to the event serializer where it is slapped on the JSON.  Since event_zip_code has a belongs_to association with event_geography, the exception_note can be accessed directly from the event_zip_code 

def exception_note
&nbsp;&nbsp;return unless trigger_exception_note == 1

&nbsp;&nbsp;event_geography.exception_note
end

This is all versus a class method which is basically a stand alone class that I would compare to active record queries you can do in the console - it knows how to do the joins and scopes, etc.  but you have to spell out all the actions.  Nikhil set up the original instance methods, I simply made modifications based on our discussion on Friday.